docs/user-guide/recipes.ipynb

```diff
@@ -0,0 +1,139 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d7608bb4-d7ce-4a36-8021-f1a52ad27ec5",
+   "metadata": {},
+   "source": [
+    "# Recipes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fce21908-20af-4580-afc7-08aaa152e828",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import NewType\n",
+    "\n",
+    "import sciline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0835a63d-f22a-477e-b887-9cdba171d8f0",
+   "metadata": {},
+   "source": [
+    "## Side effects\n",
+    "\n",
+    "It is strongly discouraged to use [side effects](https://en.wikipedia.org/wiki/Side_effect_%28computer_science%29) in code that runs as part of a pipeline.\n",
+    "This applies to, among others, file output, setting global variables, or communicating over a network.\n",
+    "The reason is that side effects rely on code running in a specific order.\n",
+    "But pipelines in Sciline have a relaxed notion of time in that the scheduler determines when and if a provider runs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8abc6b15-7749-4247-9af0-84132ef81da1",
+   "metadata": {},
+   "source": [
+    "### File output\n",
+    "\n",
+    "Files typically only need to be written at the end of a pipeline.\n",
+    "We can use [Pipeline.bind_and_call](../generated/classes/sciline.Pipeline.rst#sciline.Pipeline.bind_and_call) to call a function which writes the file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d39f143-c9da-4a7c-9ec8-ea7697b68b1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
```

Comment:
Unused? Did you plan to use it for the filename?

Reply:
Right, I used real files initially, removing.